docs/reference/index.rst

```diff
@@ -11,6 +11,7 @@ Reference Documentation
    :maxdepth: 2
 
    configuration
+   regions
```

Comment:
Should this really be top level? It seems more a subtopic of configuration. In working with this package, users will typically want to see functions related clients, solvers, etc, but typically they'll do something with region functionality once as part of setting up together with configuration.

Reply:
I thought our docs layout follows package hierarchy. And `dwave.cloud.regions` is next to `dwave.cloud.config`...

Reply:
We should balance that consideration (maintainability & documenting the code structure) with usability in terms of what pages are most accessed (operations that are repeated versus one-time configurations)

Reply:
Let's do that in a docs update (long overdue for this package).